src/anomalib/data/utils/path.py

```diff
@@ -98,3 +99,119 @@ def _resolve_path(folder: str | Path, root: str | Path | None = None) -> Path:
         # root provided; prepend root and return absolute path
         path = (Path(root) / folder).resolve()
     return path
+
+
+def contains_null_bytes(path: str | Path) -> bool:
+    r"""Check if the path contains null bytes.
+
+    Args:
+        path (str | Path): Path to check.
+
+    Returns:
+        bool: True if the path contains null bytes, False otherwise.
+
+    Examples:
+        >>> contains_null_bytes("./datasets/MVTec/bottle/train/good/000.png")
+        False
+
+        >>> contains_null_bytes("./datasets/MVTec/bottle/train/good/000.png\0")
+        True
+    """
+    return "\0" in str(path)
+
+
+def validate_path(path: str | Path, base_dir: str | Path | None = None) -> Path:
```

Comment:
I can't think of any other name but I would expect `validate_path` to return a bool. How about `valid_path`?

Reply:
hm, I would expect valid_path to return bool :) maybe `validate_and_return_path`, but a bit verbose? 

@djdameln, any thoughts on this ?

Reply:
It's a bit unclear to me what is the difference between `validate_path` and `validate_and_resolve_path`, because `validate_path` also seems to resolve the path from relative to absolute. Are both really needed, or could we reduce these two one single function called `validate_and_resolve_path`?

Reply:
validate_path resolves it to avoid accessing any other directory beyond `base_dir`? I created `validate_and_resolve_path` to not mess with the previous behaviour of `_resolve_path`